### ISSUE TITLE
Coalesce decomposed Attention pattern into single Attention op in ONNX import

### ISSUE THREAD

**Issue by antimora** [tracel-ai/burn#4107]
### Feature description

The ONNX importer in burn-import decomposes transformer Attention functionality into lower-level ops (MatMul, Softmax, Transpose, etc.) due to current exporter and model limitations (see attached screenshot for a typical pattern).

Request: Implement pattern recognition and coalescing in burn's ONNX IR to automatically detect and fuse known Attention patterns into a single Attention op. Align this with the ONNX spec (e.g., ScaledDotProductAttention operator, OpSet 21+) and route to Burn's optimized nn Attention. This will enable performance improvements (Flash Attention, fusion, etc.) and ensure ongoing ecosystem parity.

### Feature motivation

- Current models export Attention as decomposed ops due to exporter limits and data-dependent shapes (RF-DETR, transformer.py)
- ONNX ScaledDotProductAttention is unavailable in legacy exporter (OpSet ≤20); dynamo exporter (OpSet 21+) fails for some models, so pattern-based fusion in onnx-ir could help  
- Enables burn-import to leverage Burn's optimized attention for both performance and maintainability
- Potential to match other frameworks' (TensorRT) fusion capabilities

### (Optional) Suggest a Solution

- Extend burn-import's onnx-ir to coalesce decomposed Attention graph patterns into single Attention ops  
- Take inspiration from Linear fusion in onnx-ir  
- Document and test pattern recognition on models like RF-DETR  
- Retrofit to use Burn's nn Attention and enable hardware fusion (Flash Attention)

---
Additional Context:
- Related Issue: [Update ONNX op in burn-import to use Burn's nn Attention](https://github.com/tracel-ai/burn/issues/4103)
- Example model graph attached below showing decomposed attention ops

<img width="50%" height="50%" alt="Image" src="https://github.com/user-attachments/assets/28f3aef5-3010-42f8-9531-e05fdd71ec81" />

References:
- ONNX Attention Operator: https://onnx.ai/onnx/operators/onnx__Attention.html
- PyTorch ONNX Export Discussion: https://github. com/pytorch/pytorch/issues/149662
- Discord discussion: See exported notes for technical background and performance motivation.
